### ISSUE TITLE
Too much time in ISR

### ISSUE THREAD

**Issue by gin66** [gin66/FastAccelStepper#31]
Time measurement with simavr has revealed critical timing under high speed with two or three parallel steppers.

```
Two steppers: 3720us spent in ISR, while repetition rate is 4096us

Three steppers: 3806us spent in ISR, while repetition rate is 4096us
```
At 25kSteps/s within 4096us, there will be 102 stepper interrupt/s created for one stepper and 306 stepper interrupts for three steppers. With max. 6us per Interrupt, during 4096us approx 1836us will be spent in the stepper interrupt.

Possible improvement:
1.) Update README to ensure not all steppers are running simultaneously at high speed
2.) Reduce time spent in stepper interrupt
3.) Reduce time spent in fill queue ISR
4.) Reduce max. step rate for 2560
5.) Reduce max. step rate dynamically from amount of steppers used.

**Comment by gin66** [comment]
With `M1 A10000 V40 R32000 M2 A10000 V40 R32000 M3 A1000 V40 R32000` the periodicity of the fill queue ISR is not ensured and the max time in ISR has ramped up to 16ms (most likely the one or other queue has run out of commands, already)

No issue, if only two steppers are operating.

**Comment by gin66** [comment]
Had to fix bugs in the simavr-simulator.
328p runs stable, but is at the limit of the CPU.
2560 with three steppers may not work. Max_time in FillISR=18751 us, thereof <<4096us is acceptable

**Comment by gin66** [comment]
Fixed in 0.18.5
